### PR TITLE
[docs] Batch18 most functions in other

### PR DESCRIPTION
**Pull request by edgar2020**
### Description

add examples for functions `CASE`(Simple),  `CASE`(Searched), `CAST`, `COALESCE`,  `NULLIF`, and `NVL`

The PR adds examples for functions

- [X] been self-reviewed.
- [X] Link-linter passed
- [X] Spellchecker passed

### CODE REVIEW

File: docs/querying/sql-functions.md [hunk 1]
```diff
@@ -625,20 +625,87 @@ Returns the following:
 
 ## CASE
 
-`CASE expr WHEN value1 THEN result1 \[ WHEN value2 THEN result2 ... \] \[ ELSE resultN \] END`
+Returns a result based on given conditions.
```
Comment (ektravel) [inline docs/querying/sql-functions.md]:
```suggestion
Returns a result based on given conditions:
* Simple CASE compares an expression to a set of values or expressions.
* Searched CASE evaluates a set of Boolean expressions.
```

File: docs/querying/sql-functions.md [hunk 2]
```diff
@@ -625,20 +625,87 @@ Returns the following:
...
-**Function type:** [Scalar, other](sql-scalar.md#other-scalar-functions)
+* **Syntax:** `CASE expr WHEN value1 THEN result1 \[ WHEN value2 THEN result2 ... \] \[ ELSE resultN \] END`
+* **Syntax:** `CASE WHEN boolean_expr1 THEN result1 \[ WHEN boolean_expr2 THEN result2 ... \] \[ ELSE resultN \] END`
+* **Function type:** Scalar, other
 
-Returns a result based on a given condition.
+<details><summary>Examples</summary>
 
-## CAST
+The following example returns a string representing the UI type based on the value of `agent_category` from the `kttm` datasource.
```
Comment (ektravel) [inline docs/querying/sql-functions.md:638]:
```suggestion
The following example returns the UI type corresponding to the value of `agent_category` from the `kttm` datasource.
```

File: docs/querying/sql-functions.md [hunk 3]
```diff
@@ -625,20 +625,87 @@ Returns the following:
...
 
-**Function type:** [Scalar, other](sql-scalar.md#other-scalar-functions)
+Returns the following:
+
+| `device_type` | `UI_type` |
+| -- | -- |
+| `Personal computer` | `Large UI` |
+| `Smartphone` | `Mobile UI` |
+
+The following example returns a string based on the value of the `OriginStateName` column from the `flight-carriers` datasource.
```
Comment (ektravel) [inline docs/querying/sql-functions.md]:
```suggestion
The following example returns the departure location corresponding to the value of the `OriginStateName` column from the `flight-carriers` datasource.
```

File: docs/querying/sql-functions.md [hunk 4]
```diff
@@ -625,20 +625,87 @@ Returns the following:
...
+## CAST
 
 Converts a value into the specified data type.
 
+* **Syntax:** `CAST(value AS TYPE)`
+* **Function type:** Scalar, other
+
+<details><summary>Example</summary>
+
+The following example converts the values in the `Distance` column from the `flight-carriers` datasource from `DOUBLE` into type `VARCHAR`
```
Comment (ektravel) [inline docs/querying/sql-functions.md]:
```suggestion
The following example converts the values in the `Distance` column from the `flight-carriers` datasource from `DOUBLE` to `VARCHAR`.
```

File: docs/querying/sql-functions.md [hunk 5]
```diff
@@ -715,11 +782,27 @@ Alias for [`LENGTH`](#length).
...
-`COALESCE(expr, expr, ...)`
+Returns the first non-null value.
+* **Syntax:** `COALESCE(expr, expr, ...)`
+* **Function type:** Scalar, other
 
-**Function type:** [Scalar, other](sql-scalar.md#other-scalar-functions)
+<details><summary>Example</summary>
 
-Returns the first non-null value.
+The following example returns the first non-null value from `null`, `null`, `5`, and `abc`.
```
Comment (ektravel) [inline docs/querying/sql-functions.md]:
```suggestion
The following example returns the first non-null value from the list of parameters.
```

File: docs/querying/sql-functions.md [hunk 6]
```diff
@@ -2105,19 +2188,62 @@ Divides the rows within a window as evenly as possible into the number of tiles,
...
-`NVL(e1, e2)`
+Returns `value1` if `value1` is not null, otherwise returns `value2`.
 
-**Function type:** [Scalar, other](sql-scalar.md#other-scalar-functions)
+* **Syntax:** `NVL(value1, value1)`
+* **Function type:** Scalar, other
+
+<details><summary>Example</summary>
+
+The following example returns "No tail number" if the `Tail_Number` column from the `flight-carriers` datasource is null.
```
Comment (ektravel) [inline docs/querying/sql-functions.md]:
```suggestion
The following example replaces each null value in the `Tail_Number` column of the `flight-carriers` datasource with the string "No tail number."
```

File: docs/querying/sql-scalar.md [hunk 7]
```diff
@@ -276,7 +276,7 @@ The [DataSketches extension](../development/extensions-core/datasketches-extensi
 |`CASE expr WHEN value1 THEN result1 \[ WHEN value2 THEN result2 ... \] \[ ELSE resultN \] END`|Simple CASE.|
 |`CASE WHEN boolean_expr1 THEN result1 \[ WHEN boolean_expr2 THEN result2 ... \] \[ ELSE resultN \] END`|Searched CASE.|
 |`CAST(value AS TYPE)`|Cast value to another type. See [Data types](sql-data-types.md) for details about how Druid SQL handles CAST.|
-|`COALESCE(value1, value2, ...)`|Returns the first value that is neither NULL nor empty string.|
+|`COALESCE(value1, value2, ...)`|Returns the first value not NULL.|
```
Comment (ektravel) [inline docs/querying/sql-scalar.md]:
```suggestion
|`COALESCE(value1, value2, ...)`|Returns the first non-null value.|
```

**Review by ektravel — COMMENTED** [review]
Updated text to improve readability. Tested all of the code examples.

**Review by techdocsmith — COMMENTED** [review]
LGTM

**Pull request merged**